packages/eslint-plugin/test/no-deprecated-type-references.test.ts

```diff
@@ -0,0 +1,153 @@
+                    messageId: "migration",
+                    data: { deprecatedTypeName: "IProps", newTypeName: "Props" },
+                },
+            ],
+            output: dedent`
+                import { Props } from "@blueprintjs/core";
+
+                export interface MyInterface extends Props {
+                    foo: string;
+                }
+            `,
+        },
+        {
+            code: dedent`
+                import * as Blueprint from "@blueprintjs/core";
+
+                export interface MyInterface extends Blueprint.IProps {
+                    foo: string;
+                }
+            `,
+            errors: [
+                {
+                    messageId: "migration",
+                    data: { deprecatedTypeName: "IProps", newTypeName: "Props" },
+                },
+            ],
+            output: dedent`
+                import * as Blueprint from "@blueprintjs/core";
+
+                export interface MyInterface extends Blueprint.Props {
+                    foo: string;
+                }
+            `,
+        },
+        {
+            code: dedent`
+                import { MenuItem } from "@blueprintjs/core";
+                import { IItemRendererProps } from "@blueprintjs/select";
+
+                export const defaultRenderMenuItem = (_item: any, { handleClick, modifiers }: IItemRendererProps) => {
+                    return <MenuItem {...modifiers} onClick={handleClick} />;
+                };
+            `,
+            errors: [
+                {
+                    messageId: "migration",
+                    data: { deprecatedTypeName: "IItemRendererProps", newTypeName: "ItemRendererProps" },
+                },
+            ],
+            output: dedent`
+                import { MenuItem } from "@blueprintjs/core";
+                import { ItemRendererProps } from "@blueprintjs/select";
+
+                export const defaultRenderMenuItem = (_item: any, { handleClick, modifiers }: ItemRendererProps) => {
+                    return <MenuItem {...modifiers} onClick={handleClick} />;
+                };
+            `,
+        },
+        // N.B. it is difficult to test multiple fixes, so we test those separately
+        {
+            code: dedent`
+                import { Select } from "@blueprintjs/select";
+                import { ITimezoneItem } from "@blueprintjs/timezone";
+                const MySelect = Select.ofType<ITimezoneItem>();
+            `,
+            errors: [
+                {
+                    messageId: "migration",
+                    data: { deprecatedTypeName: "ITimezoneItem", newTypeName: "TimezoneItem" },
+                },
+            ],
+            output: dedent`
+                import { Select } from "@blueprintjs/select";
+                import { TimezoneItem } from "@blueprintjs/timezone";
+                const MySelect = Select.ofType<TimezoneItem>();
+            `,
+        },
+        {
```

Comment:
can you add a test for when there are multiple imports from the same package?

Reply:
I tried to do this, but for some reason the rule tester fixer doesn't like to run multiple auto-fixes. so when I write a test like this:

```tsx
                import { IItemRendererProps, ISelectProps } from "@blueprintjs/select";
                const mySelectProps: ISelectProps = { items: [], renderItem };
                const renderItem = (_item: any, { handleClick, modifiers }: IItemRendererProps) => {
                    return <MenuItem {...modifiers} onClick={handleClick} />;
                };
```

expecting it to be autofixed to:

```tsx
                import { ItemRendererProps, SelectProps } from "@blueprintjs/select";
                const mySelectProps: SelectProps = { items: [], renderItem };
                const renderItem = (_item: any, { handleClick, modifiers }: ItemRendererProps) => {
                    return <MenuItem {...modifiers} onClick={handleClick} />;
                };
```

then it doesn't pass, since the rule tester only runs the first auto-fix. I don't want to write a test with weird expected output showing only one of the deprecated type symbols being fixed. I can look into if this is a bug on ESLint's part...

Reply:
is it possible to test without the fixer, but show that both symbols are flagged as errors instead?

Reply:
discussed offline, ESLint does not allow testing fixable rules without running their fixers. probably another FR for ESLint given the limitations of fixer output in RuleTester (see https://github.com/eslint/eslint/issues/11187#issuecomment-470990425)